packages/cli/src/ampx.ts

```diff
@@ -32,11 +35,11 @@ attachUnhandledExceptionListeners(usageDataEmitter);
 
 verifyCommandName();
 
-const parser = createMainParser(libraryVersion, usageDataEmitter);
-
-await parseAsyncSafely(parser);
```

Comment:
Is this change related to the yargs 'middleware' error handling? I'm trying to understand the logic behind this refactor.

Reply:
Yes, see below https://github.com/aws-amplify/amplify-backend/blob/b90354a6ecc58ae92d088e9c204af85e5ef486ec/packages/cli/src/ampx.ts#L37-L47

Here on line 37, we are always "succeeding" no matter if the command was successful or not causing the line 47 to be executed always sending the "Success" metric. I don't really see any value in adding the error handling middleware if we have a top level parser and we add try catch there.